app.py

```diff
@@ -21,6 +22,24 @@ def index():
                 print(f'{c.id}: {c.text}')
 
 
+def index_self_data(data_dir):
```

Comment:
`index()` and `index_self_data` is redundant. 

```suggestion
def index(data_dir):
```